custom_components/unifiprotect/unifi_protect_server.py

```diff
@@ -40,293 +53,248 @@ def __init__(
+                                "type": str(camera["type"]),
+                                "recording_mode": recording_mode,
+                                "ir_mode": ir_mode,
+                                "rtsp": rtsp,
+                                "up_since": upsince,
+                                "last_motion": lastmotion,
+                                "online": online,
+                                "motion_start": None,
+                                "motion_score": 0,
+                                "motion_thumbnail": None,
+                                "motion_on": False,
+                            }
                         }
-                    }
-                    self.device_data.update(item)
-                else:
-                    camera_id = camera["id"]
-                    self.device_data[camera_id]["last_motion"] = lastmotion
-                    self.device_data[camera_id]["online"] = online
-                    self.device_data[camera_id]["up_since"] = upsince
-                    self.device_data[camera_id]["recording_mode"] = recording_mode
-                    self.device_data[camera_id]["ir_mode"] = ir_mode
-        else:
-            raise NvrError(
-                "Fetching Camera List failed: %s - Reason: %s"
-                % (response.status_code, response.reason)
-            )
+                        self.device_data.update(item)
+                    else:
+                        camera_id = camera["id"]
+                        self.device_data[camera_id]["last_motion"] = lastmotion
+                        self.device_data[camera_id]["online"] = online
+                        self.device_data[camera_id]["up_since"] = upsince
+                        self.device_data[camera_id]["recording_mode"] = recording_mode
+                        self.device_data[camera_id]["ir_mode"] = ir_mode
+            else:
+                raise NvrError(
+                    f"Fetching Camera List failed: {response.status} - Reason: {response.reason}"
+                )
 
-    def _get_motion_events(self, lookback=86400):
+    async def _get_motion_events(self, lookback: int = 86400) -> None:
         """Load the Event Log and loop through items to find motion events."""
-
         event_start = datetime.datetime.now() - datetime.timedelta(seconds=lookback)
         event_end = datetime.datetime.now() + datetime.timedelta(seconds=10)
         start_time = int(time.mktime(event_start.timetuple())) * 1000
         end_time = int(time.mktime(event_end.timetuple())) * 1000
 
-        event_uri = (
-            "https://"
-            + str(self._host)
-            + ":"
-            + str(self._port)
-            + "/api/events?end="
-            + str(end_time)
-            + "&start="
-            + str(start_time)
-            + "&type=motion"
-        )
-
-        response = self.req.get(
+        event_uri = f"{self._base_url}/api/events"
```

Comment:
Also updated some of these GET requests to pass the params, rather than putting them in the url query string.  It's a bit safer this way as they will get encoded properly by aiohttp.